### PR TITLE
Show --noImplicitAny as an option to throw errors, not warnings

### LINKED ISSUE #1632: --noImplicitAny is an error, not a warning
The help says it's a warning, but it actually fails the build with an error:

    --noImplicitAny    Warn on expressions and declarations with an implied 'any' type.

I believe this should say something like:

    --noImplicitAny    Raise an error on expressions and declarations with an implied 'any' type.

### PR DESCRIPTION
**Pull request by lptr**
Fixes #1632

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
:+1:

(Also, I severely want to be friends with MSBOT now.)

File: src/services/resources/diagnosticMessages.json [hunk 1]
```diff
@@ -1743,7 +1743,7 @@
         "category": "Error",
         "code": 7003
     },
-    "Warn on expressions and declarations with an implied 'any' type.": {
+    "Raise error on expressions and declarations with an implied 'any' type.": {
```
Comment (mhegazy) [inline src/services/resources/diagnosticMessages.json]: you do not need to touch this file, it is obsolete, and will be deleted.

File: src/services/syntax/SyntaxGenerator.d.ts [hunk 2]
```diff
@@ -417,7 +417,7 @@ declare module TypeScript {
         This_version_of_the_Javascript_runtime_does_not_support_the_0_function: string;
         Unknown_rule: string;
         Invalid_line_number_0: string;
-        Warn_on_expressions_and_declarations_with_an_implied_any_type: string;
+        Raise_error_on_expressions_and_declarations_with_an_implied_any_type: string;
```
Comment (mhegazy) [inline src/services/syntax/SyntaxGenerator.d.ts]: this one as well.

File: src/services/compiler/tsc.ts [hunk 3]
```diff
@@ -413,7 +413,7 @@ module TypeScript {
 
             opts.flag('noImplicitAny', {
                 usage: {
```
Comment (mhegazy) [inline src/services/compiler/tsc.ts]: you do not need to touch this file, it is obsolete, and will be deleted.

File: src/services/resources/diagnosticCode.generated.ts [hunk 4]
```diff
@@ -436,7 +436,7 @@ module TypeScript {
         This_version_of_the_Javascript_runtime_does_not_support_the_0_function: "This version of the Javascript runtime does not support the '{0}' function.",
         Unknown_rule: "Unknown rule.",
         Invalid_line_number_0: "Invalid line number ({0})",
-        Warn_on_expressions_and_declarations_with_an_implied_any_type: "Warn on expressions and declarations with an implied 'any' type.",
+        Raise_error_on_expressions_and_declarations_with_an_implied_any_type: "Raise error on expressions and declarations with an implied 'any' type.",
```
Comment (mhegazy) [inline src/services/resources/diagnosticCode.generated.ts]: same here.

**Comment by mhegazy** [comment]
thanks @lptr.

**Pull request merged**